lib/puppet/face/module/generate.rb

```diff
@@ -129,13 +129,17 @@
       puts _("Finished; module generated in %{path}.") % { path: path }
       result.join("\n")
     end
+
+    deprecate
   end
 end
 
 module Puppet::ModuleTool::Generate
   module_function
 
   def generate(metadata, skip_interview = false)
+    Puppet.deprecation_warning _("`puppet module generate` is deprecated and will be removed in a future release. This action has been replaced by Puppet Development Kit. For more information visit https://puppet.com/docs/pdk/latest/pdk.html.")
```

Comment:
Might be worth adding a translator note here that parts of this string should not be translated and are literals.

e.g. https://github.com/puppetlabs/puppet/blob/80e7948c69b6c2145a709c6e7fc95bd7bb772f26/lib/puppet/util/command_line/trollop.rb#L815